src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,144 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node head;
+    private Node tail;
+    private int size = 0;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            head = new Node(null, value, null);
+            tail = head;
+        } else {
+            Node secondLast = tail;
+            tail = new Node(secondLast, value, null);
+            secondLast.next = tail;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Invalid index: " + index);
+        }
+
+        if (index == size) {
+            add(value);
+        } else {
```

Comment:
```suggestion
            add(value);
            return;
        }
```
better return early than wrap and indent big chunks of code

Reply:
got it